analysis/maxsavings.py

```diff
@@ -71,4 +72,48 @@ def max_dam_savings(data, system_uptime, continuous_flex, baseload):
     return flex.model.pct_cost_savings()
 
 
-# TODO - implement tariff savings for a single case
+def get_start_end(month):
+    if month == 12:
+        start_dt = np.datetime64("2024-" + str(month) + "-01")
+        end_dt = np.datetime64("2025-01-01")
+    elif month > 9:
+        start_dt = np.datetime64("2024-" + str(month) + "-01")
+        end_dt = np.datetime64("2024-" + str(month+1) + "-01")
+    elif month == 9:
+        start_dt = np.datetime64("2024-09-01")
+        end_dt = np.datetime64("2024-10-01")
+    else:
+        start_dt = np.datetime64("2024-0" + str(month) + "-01")
+        end_dt = np.datetime64("2024-0" + str(month+1) + "-01")
+
+    return (start_dt, end_dt)
+
+
+def max_tar_savings(
+    data, 
+    system_uptime, 
+    continuous_flex, 
+    baseload, 
+    startdate_dt, 
+    enddate_dt
+):
+
+    flex = flexloadMILP(
+        baseload=baseload,
+        cost_signal=data,
+        costing_type="tariff",
+        costing_path=None,
+        emissions_type=None,
+        emissions_path=None,
+        flex_capacity=continuous_flex,
+        rte=1.0,
+        min_onsteps=max(int(len(data) * (system_uptime)), 1),
+        uptime_equality=True,
```

Comment:
I think the tariff function needs to have an arg for `uptime_equality`. In the contour plot we want this to be `True` since we're sweeping over all the up-times but in the box plot we want to optimize for this so uptime equality should be `False`.